structure/fenwicktree/fenwicktree.go

```diff
@@ -0,0 +1,60 @@
+// Fenwick Tree Data Structure for efficient range queries on an array of integers.
+// Also known as Binary Indexed Tree. It can query the sum of any range of the array and
+// can update the array at a specific position by adding a value to it (point update).
+// Build: O(N)
+// Query: O(log(N))
+// Update: O(log(N))
+// reference: https://brilliant.org/wiki/fenwick-tree/
+package fenwicktree
+
+// FenwickTree represents the data structure of the Fenwick Tree
+type FenwickTree struct {
+	n     int   // n: Size of the input array.
+	array []int // array: the input array on which queries are made.
+	bit   []int // bit: store the sum of ranges.
```

Comment:
These comments are usually redundant if the naming of the fields is chosen correctly.